### DISCUSSION TITLE
Expanding list of supported currencies: how to proceed?

### DISCUSSION THREAD

**Discussion by skrattaren** [sharkdp/numbat#553] [Q&A] [answered]
Hello and thank for the great tool, first of all.

Numbat already supports pretty long, but not exhaustive list of currencies, and I personally use several others on the daily basis.  I looked through source code, PRs and issues, and it seems to me that the list is pretty arbitrary (which is fine), and adding more currencies to `currencies.nbt` should not be a problem (until *#438 [Currency providers]* is implemented, at least).  What would be your advice?  Should I create a trivial PR adding the currencies I need, or would it be better to stick to a local plugin / `init.nbt`?

**Comment by sharkdp** [comment, marked as answer]
Hi, thank you for the feedback.

> it seems to me that the list is pretty arbitrary

The list is not arbitrary, it's based on the currencies that are available in the ECB list that we [use in Numbat](https://github.com/sharkdp/numbat/blob/21c5b6709e899da5ea8efcd474a71b28d849b6a7/numbat-exchange-rates/src/lib.rs#L37):

https://www.ecb.europa.eu/stats/eurofxref/eurofxref-daily.xml

> and adding more currencies to `currencies.nbt` should not be a problem

We can add additional currencies, but the exchange rates will not be available. The `exchange_rate` function does not yield an error, but returns `NaN` instead. This is intentional to make currency units available even if there is no internet connection. If you add additional currencies that are not in the ECB list, they could be used as units.. and combined with other units etc. But you can not use them for currency conversion operations.

With that in mind, I'm definitely in favor of adding even more currencies, even if the exchange rates are not (yet) available for them (the latter will be addressed in #438)

**Reply by skrattaren** [reply]
Thank you for the prompt response and extensive commentary, much appreciated!

So to summarize:
- PR adding other currencies is welcome, even for an arbitrary list of currencies some random guy (me) needs
- conversion wouldn't work for them, though
- following #438 (and maybe helping if possible) is recommended.

Have I got this right?

**Reply by sharkdp** [reply]
Yes, exactly.
